config/routes.rb

```diff
@@ -154,6 +154,7 @@
     post "/jenkins/:token" => "jenkins#create", as: :jenkins_deploy
     post "/buildkite/:token" => "buildkite#create", as: :buildkite_deploy
     post "/github/:token" => "github#create", as: :github_deploy
+    post "/generic_integration/:token" => "generic_integration#create", as: :generic_integration_deploy
```

Comment:
can you consider having different URL? 
/integrations/generic_integration little awkward,  maybe only /integrations/generic should be fine.